test/modules/amqp.py

```diff
@@ -40,6 +44,14 @@ def on_message(self, body, message):
     def on_conn_retry(self):
         LOG.error('Retrying connection for {0}'.format(self.__amqp_url))
 
+    def on_connection_error(self, exc, interval):
```

Comment:
Good point.  Maybe I can call ensure_connection() and reset __max_error if the connection good?